### PR TITLE
#130-store-page

### PR DESCRIPTION
**Pull request by AbdelrahmanMohyeldin**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/app/domain/product/product.repository.ts [hunk 1]
```diff
@@ -21,8 +23,8 @@ export class ProductRepository extends ResourceService<Product> {
   getResourceUrl(): string {
     return 'product';
   }
-  filterProductsById(id:number):Observable<Response<Category>>{
-    return this.categoryRepository.getListById(`${id}/product`)
+  filterProductsById(id:number): Observable<Response>{
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/product/product.repository.ts:26]: where is return type?

File: retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        this.translate.instant('store.close'),
+        {
+          duration: 2000,
+        }
+      ); 
+    }
+  })
+ }
+
+  storgeForm(){
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts:74]: is this meaningful name?

File: retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === 'yes') {
+        this.deleteStore(store);
+      }
+    });
+  }
+
+  getAllEmployee(): void {
+    this.employeeRepository.getList().subscribe((result) => {
+      this.employees = result.data;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts:186]: where is pagination?

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

**Comment by ZinabHashem** [comment]
image

**Comment by ZinabHashem** [comment]
height is very big make it dynamic 
image

**Comment by ZinabHashem** [comment]
i need to do two select to call filter function 
test this and fix plz

File: retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts (hunk 3 above)
Comment (AbdelrahmanMohyeldin) [reply retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts:186]: I think I don't need pagination as I just use getAllEmployee() to display data in select form field in the form

**Comment by AbdelrahmanMohyeldin** [comment]
> height is very big make it dynamic image

you mean the height of the filter input?

File: retail-managment-system/rms-ui/src/app/domain/store/store.repository.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(httpclient:HttpClient) {
+    super(httpclient)
+   }
+
+   getResourceUrl(): string {
+    return 'store'
+ }
+
+ filterStoreByEmployee(page:number,size:number,responsible:string):Observable<Response<Store>>{
+  return this.getList({page,size,responsible})
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/store/store.repository.ts:22]: call getList direct in component not need to do this function

**Comment by ZinabHashem** [comment]
> > height is very big make it dynamic image
> 
> you mean the height of the filter input?


i mean this height
image

File: retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { Component } from '@angular/core';
   selector: 'app-side-bar',
   templateUrl: './side-bar.component.html',
   styles: [
-    '.active { background-color: #002d40; color: white; padding: 10px;} .mat-divider { border-top-width: 3px; border-color: #002d40; }',
+    '.active-link { background-color: #002d40; color: white; padding:10px; width:fit-content} .active { background-color: #002d40; color: white; padding: 10px;} .mat-divider { border-top-width: 3px; border-color: #002d40;} .mat-expansion-panel-header{padding:0 57px;} .mat-expansion-panel:not(.mat-expanded) .mat-expansion-panel-header:hover:not([aria-disabled=true]) {background:none;}',
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.ts:7]: use style sheet if you will write alot of style

File: retail-managment-system/rms-ui/src/app/pages/store/store/search-store/search-store.component.scss [hunk 6]
```diff
@@ -0,0 +1,4 @@
+  th,
+  td {
+    line-height: 2;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/store/store/search-store/search-store.component.scss:3]: if this just style 
so delete this file and add style in ts

File: retail-managment-system/rms-ui/src/app/pages/store/store/search-store/search-store.component.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/store/store/search-store/search-store.component.ts:1]: there are two store folder delete one

**Pull request merged**